imap_processing/spice/geometry.py

```diff
@@ -443,3 +444,74 @@ def instrument_pointing(
     if isinstance(et, typing.Collection):
         return np.rad2deg([spice.reclat(vec)[1:] for vec in pointing])
     return np.rad2deg(spice.reclat(pointing)[1:])
+
+
+def cartesian_to_spherical(
+    v: NDArray,
+) -> NDArray:
+    """
+    Convert cartesian coordinates to spherical coordinates.
+
+    Parameters
+    ----------
+    v : np.ndarray
+        A NumPy array with shape (n, 3) where each
+        row represents a vector
+        with x, y, z-components.
+
+    Returns
+    -------
+    spherical_coords : np.ndarray
+        Spherical coordinate.
```

Comment:
Can you define the order of coordinates output and their ranges here? Could lift some of the documentation from spice: https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/recsph_c.html